mzLib/Omics/BaseSpectralMatch.cs

```diff
@@ -0,0 +1,378 @@
+﻿namespace Omics
+{
+    /// <summary>
+    /// A spectral match implementation that supports fragment-level sequence coverage calculation.
+    /// Implements both <see cref="ISpectralMatch"/> for basic PSM functionality and 
+    /// <see cref="IHasSequenceCoverageFromFragments"/> for determining which residues are 
+    /// covered by matched fragment ions.
+    /// 
+    /// Coverage is determined by analyzing N-terminal and C-terminal fragment positions to identify
+    /// residues that have fragment evidence on both sides. This works for any biopolymer type
+    /// (peptides use b/y ions, nucleic acids use 5'/3' fragments).
+    /// </summary>
+    public class BaseSpectralMatch : ISpectralMatch, IHasSequenceCoverageFromFragments, IEquatable<BaseSpectralMatch>
+    {
+        private readonly List<IBioPolymerWithSetMods> _identifiedBioPolymers;
```

Comment:
Remove this. When we need the identified biopolymers, use the ISpectralMatch.GetIdentifiedBioPolymersWithSetMods() method